src/main/java/gregtech/api/capability/impl/FuelRecipeLogic.java

```diff
@@ -111,18 +110,32 @@ public int getNetworkID() {
     @Override
     public void update() {
         if (getMetaTileEntity().getWorld().isRemote) return;
+
         if (workingEnabled) {
-            if (recipeDurationLeft > 0) {
-                if (energyContainer.get().getEnergyCanBeInserted() >=
-                    recipeOutputVoltage || shouldVoidExcessiveEnergy()) {
-                    energyContainer.get().addEnergy(recipeOutputVoltage);
-                    if (--this.recipeDurationLeft == 0) {
-                        this.wasActiveAndNeedsUpdate = true;
-                    }
-                }
+
+            /* we need to compute those two before doing anything
+             * to avoid loosing 1 tick of production
+             */
+
+            boolean canProduce = canProduceEnergy() &&
+                    (energyContainer.get().getEnergyCanBeInserted() >= calculateRecipeOutputVoltage() ||
+                            shouldVoidExcessiveEnergy());
+
+            boolean canConsume = canConsumeFuel() && (shouldConsumeWhenNotProducing() || canProduce);
```

Comment:
Both of these seem to be bad abstractions to build logic upon. Why do you even have logic additionally to calling canProduceEnergy here? You should literally move it to default implementation of canProduceEnergy and just use canProduceEnergy without doing any additional checks

Same applies to canConsumeFuel

Reply:
No, because then the subclass then has to reproduce these implementation details to override the policy and will get out of sync when new policy is added to the default implementation.

Reply:
You realize you can just call super method to follow parent policy and then && with your custom condition, right?
Hard-coding conditions like that just to force parent implementation is stupid.

Reply:
read warjot comment please this has implemented like this because he pointed out that doing this in the way you want was messy and I agree

Reply:
>You realize you can just call super method to follow parent policy and then && with your custom condition, right?
Hard-coding conditions like that just to force parent implementation is stupid.

It's not stupid it's clearer you can easily change canProduce without minding you can change the voiding policy without minding or risking affecting the other policy

Reply:
Only for simple overrides. 
If the default implementation is doing 5 steps and I want to change logic at step 3, super isn't going to help.
The user should not have to reproduce implementation details (that will change) 
for anything beyond a BEFORE or AFTER injection.
Like I said above, these policy overrides should concentrate on "doing one thing well".

Reply:
Yeah, you do not risk changing it because, you know, you cannot change it at all with your implementation?
If you consider calling super implementation such a difficult aspect to be mindful about, I have some bad news for you

Reply:
> Only for simple overrides.
If the default implementation is doing 5 steps and I want to change logic at step 3, super isn't going to help.
The user should not have to reproduce implementation details (that will change)
for anything beyond a BEFORE or AFTER injection.

Yeah, and the funniest thing is, do you actually realize *current* implementation only allows simple overrides? It literally does && with the value returned from canProduceEnergy(), you cannot change anything after this point.
Keeping it like that does not solve the issue of injecting into original logic, it literally makes it even worse because you just CANNOT change parent logic at all.

Reply:
> Yeah, and the funniest thing is, do you actually realize current implementation only allows simple overrides? It literally does && with the value returned from canProduceEnergy(), you cannot change anything after this point.

Not true because there are 2 policy options in play here;

canProduceEnergy()
shouldVoidExcessEnergy()

You can override these policies independently without worrying about implementation details.
You don't have worry about updating your override if somebody introduces a 3rd piece of logic (possibly overriden by an intermediate class in the hierarchy).

Reply:
you can still override and not worry about implementation details in parent, granted you just call super
Like really, this logic being moved to canProduceEnergy does not change anything, but allows you to completely override it if you really need to

Reply:
If you are completely overridding in a sledgehammer way, you are probably extending the wrong class?
I still disagree about having to redo implementation details (which is fairly easy to show) but this argument could go on forever.